src/main/java/com/cloudogu/smeagol/repository/infrastructure/ScmRepositoryRepository.java

```diff
@@ -0,0 +1,58 @@
+package com.cloudogu.smeagol.repository.infrastructure;
+
+import com.cloudogu.smeagol.ScmHttpClient;
+import com.cloudogu.smeagol.repository.domain.*;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class ScmRepositoryRepository implements RepositoryRepository {
+
+    private ScmHttpClient scmHttpClient;
+
+    @Autowired
+    public ScmRepositoryRepository(ScmHttpClient scmHttpClient) {
+        this.scmHttpClient = scmHttpClient;
+    }
+
+    @Override
+    public Iterable<Repository> findAll() {
+        Optional<RepositoryDTO[]> dtos = scmHttpClient.get("/api/rest/repositories.json", RepositoryDTO[].class);
+
+        return Arrays.stream(dtos.get())
+            .filter(dto -> dto.type.equals("git"))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the "git" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1132)